File: source/qinka_computing_subprogram.c
#include <qinka_cs/qinka_computing_subprogram.h>
#include <config.h>
#include <config_gen.h>

static
const char * _qinka_computing_subprogram_ver_str = _QINKA_CS_VER_STR;
static
const char * _qinka_computing_subprogram_backend_str = _ACC_BACKEND_STR;

const char * qinka_computing_subprogram_version_str() {
    return _qinka_computing_subprogram_ver_str;
}

int qinka_computing_subprogram_version_major() {
    return _QINKA_CS_VER_MAJOR;
}

int qinka_computing_subprogram_version_minor() {
    return _QINKA_CS_VER_MINOR;
}

int qinka_computing_subprogram_version_patch() {
    return _QINKA_CS_VER_PATCH;
}

int qinka_computing_subprogram_version_diff() {
    return _QINKA_CS_VER_DIFF;
}

const char * qinka_computing_subprogram_backend_type_str() {
    return _qinka_computing_subprogram_backend_str;
}

int qinka_computing_subprogram_backend_type_id() {
    return _ACC_BACKEND_ID;
}

File: bindings/hQinkaCS/src/Accelerate/QinkaCS/QinkaComputingSubprogram.hs
module Accelerate.QinkaCS.QinkaComputingSubprogram where

import           Accelerate.QinkaCS.Binding.QinkaComputingSubprogram

qinkaComputingSubprgramVersionStr :: String
qinkaComputingSubprgramVersionStr = qinka_computing_subprogram_version_str

qinkaComputingSubprgramVersion :: (Int, Int, Int, Int)
qinkaComputingSubprgramVersion = ( qinka_computing_subprogram_version_major
                                 , qinka_computing_subprogram_version_minor
                                 , qinka_computing_subprogram_version_patch
                                 , qinka_computing_subprogram_version_diff
                                 )

qinkaComputingSubprgramBackend   :: String
qinkaComputingSubprgramBackend   = qinka_computing_subprogram_backend_type_str
qinkaComputingSubprgramBackendId :: Int
qinkaComputingSubprgramBackendId = qinka_computing_subprogram_backend_type_id
